core/services/chainlink/application.go

```diff
@@ -158,8 +159,14 @@ func (app *ChainlinkApplication) Start() error {
 		app.Exiter(0)
 	}()
 
+	err := app.Store.EthClient.Dial(context.TODO())
```

Comment:
I originally found `context.TODO()` kind of weird as well, but I ended up coming around to the Go authors' reasoning (which is similar to the reasoning behind `type X = Y`-style type aliases).  The idea is that, during a large refactor, it can be very noisy to add contexts (2-3 new lines in each place, at least), and sometimes, it takes a lot of extra thought to figure out how exactly to tune the timeouts/deadlines.  We currently have some stories in Tracker about ensuring that contexts are added appropriately, so I figured that marking all of the new contexts in this PR as "TODO" would be helpful to whomever picks up those other stories.  I don't get the sense that they'll be this way for long, but if the other work drags on, I'm happy to switch them to `Background`.